File: knowledge/public/templates/mobile-webview-handoff/android/KyberionHandoffModels.kt
package dev.kyberion.handoff

data class KyberionCookie(
    val name: String,
    val value: String,
    val domain: String? = null,
    val path: String? = "/",
    val expires: Long? = null,
    val httpOnly: Boolean? = null,
    val secure: Boolean? = null,
    val sameSite: String? = null
)

data class KyberionWebviewSessionHandoff(
    val kind: String = "webview-session-handoff",
    val target_url: String,
    val origin: String? = null,
    val cookies: List<KyberionCookie> = emptyList(),
    val local_storage: Map<String, String> = emptyMap(),
    val session_storage: Map<String, String> = emptyMap(),
    val headers: Map<String, String> = emptyMap(),
    val source: Map<String, String> = mapOf(
        "platform" to "android",
        "app_id" to "example-mobile-login-passkey"
    )
)

data class WebViewSessionState(
    val url: String,
    val origin: String? = null,
    val cookies: List<KyberionCookie> = emptyList(),
    val localStorage: Map<String, String> = emptyMap(),
    val sessionStorage: Map<String, String> = emptyMap(),
    val headers: Map<String, String> = emptyMap()
)

fun interface WebViewStateReader {
    fun readCurrentState(): WebViewSessionState
}

File: knowledge/public/templates/mobile-webview-handoff/ios/KyberionHandoffModels.swift
import Foundation

struct KyberionCookie: Codable {
    let name: String
    let value: String
    let domain: String?
    let path: String?
    let expires: Int?
    let httpOnly: Bool?
    let secure: Bool?
    let sameSite: String?
}

struct KyberionWebviewSessionHandoff: Codable {
    let kind: String
    let target_url: String
    let origin: String?
    let cookies: [KyberionCookie]
    let local_storage: [String: String]
    let session_storage: [String: String]
    let headers: [String: String]
    let source: [String: String]

    init(
        targetURL: String,
        origin: String?,
        cookies: [KyberionCookie],
        localStorage: [String: String],
        sessionStorage: [String: String],
        headers: [String: String],
        appId: String = "example-ios-login-passkey"
    ) {
        self.kind = "webview-session-handoff"
        self.target_url = targetURL
        self.origin = origin
        self.cookies = cookies
        self.local_storage = localStorage
        self.session_storage = sessionStorage
        self.headers = headers
        self.source = [
            "platform": "ios",
            "app_id": appId
        ]
    }
}

struct WebViewSessionState {
    let url: URL
    let origin: String?
    let cookies: [KyberionCookie]
    let localStorage: [String: String]
    let sessionStorage: [String: String]
    let headers: [String: String]
}

protocol WebViewSessionReader {
    func readCurrentState() throws -> WebViewSessionState
}
